### ISSUE TITLE
Uncaught TypeError: Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by Kame7C0** [fand/veda#19]
[Enter steps to reproduce:]

1. Open Atom >> ctrl+shift+p >> glsl livecoder toggle; to enable
2.  ctrl+shift+p >> glsl livecoder toggle; to disable (and get this error)... you cant enable it again

**Atom**: 1.20.0 x64 also tested in 1.19.6 x64
**Electron**: 1.6.9
**OS**: Ubuntu 17.04
**Thrown From**: [glsl-livecoder](https://github.com/fand/glsl-livecoder) package 0.9.2

### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /home/ck/.atom/packages/glsl-livecoder/lib/camera-loader.js:40

TypeError: Cannot read property 'then' of undefined
    at CameraLoader.disable (/packages/glsl-livecoder/lib/camera-loader.js:40:19)
    at ThreeShader.stop (/packages/glsl-livecoder/lib/three-shader.js:246:24)
    at GlslLivecoder.stop (/packages/glsl-livecoder/lib/glsl-livecoder.js:141:17)
    at GlslLivecoder.toggle (/packages/glsl-livecoder/lib/glsl-livecoder.js:127:14)
    at HTMLElement.glslLivecoderToggle (/packages/glsl-livecoder/lib/index.js:95:47)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:265:35)
    at /usr/share/atom/resources/app/src/command-registry.js:3:65
    at Object.didConfirmSelection (/usr/share/atom/resources/app/node_modules/command-palette/lib/command-palette-view.js:106:35)
    at SelectListView.confirmSelection (/usr/share/atom/resources/app/node_modules/atom-select-list/src/select-list-view.js:313:26)
    at HTMLDivElement.core:confirm (/usr/share/atom/resources/app/node_modules/atom-select-list/src/select-list-view.js:81:20)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:265:35)
    at /usr/share/atom/resources/app/src/command-registry.js:3:65
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -0:26.1.0 intentions:highlight (input.hidden-input)
     -0:26 command-palette:toggle (input.hidden-input)
     -0:23.9.0 core:confirm (input.hidden-input)
     -0:23.4.0 glsl-livecoder:toggle (input.hidden-input)
     -0:22.2.0 intentions:highlight (input.hidden-input)
     -0:22 glsl-livecoder:load-shader (input.hidden-input)
     -0:21 intentions:highlight (input.hidden-input)
     -0:19.9.0 command-palette:toggle (input.hidden-input)
     -0:18.7.0 core:confirm (input.hidden-input)
     -0:18.7.0 glsl-livecoder:toggle (input.hidden-input)
```

### Non-Core Packages

```
atom-rst-preview-docutils 1.0.0 
atom-typescript 11.0.8 
autocomplete-glsl 0.2.3 
autocomplete-python 1.10.2 
busy-signal 1.4.3 
glsl-livecoder 0.9.2 
intentions 1.1.5 
javascript-snippets 1.2.1 
language-glsl 2.0.1 
language-ini 1.19.0 
language-javascript-jsx 0.3.7 
language-kivy 0.2.0 
language-restructuredtext 1.1.0 
linter 2.2.0 
linter-glsl 2.1.3 
linter-ui-default 1.6.8 
MagicPython 1.0.12 
minimap 4.29.6 
minimap-find-and-replace 4.5.2 
open-unsupported-files 1.0.20 
pretty-json 1.6.4 
python-indent 1.1.0 
python-tools 0.6.9 
run-in-terminal 1.0.1 
script 3.15.0 
theme-switcher 1.1.0 
videoplayer 1.2.1 
```

**Comment by Kame7C0** [comment]
0.9.1 broken 0.9.0 broken
0.8.1 works :smiley:

**Comment by fand** [comment]
The bug was caused by v0.9.0 😓 
I fixed and published v0.9.3, thanks!

**Comment by Kame7C0** [comment]
that was fast
